### DISCUSSION TITLE
Editing related fields in a form

### DISCUSSION THREAD

**Discussion by kolaente** [filamentphp/filament#5922] [Help]
My users have a `personalData` relation which, among other things, allows them to save an address. I was able to show the address in a table column with the `Tables\Columns\TextColumn::make('personalData.street')` definition. However, the same thing does not work for a text input, like this: `Forms\Components\TextInput::make('personalData.street')` - I only get an empty field saying "Street".

What am I doing wrong? The docs seem to only explain how to use relationships for select fields.

**Comment by kolaente** [comment]
I got something working by [putting the related model data manually into $data](https://filamentphp.com/docs/2.x/admin/resources/editing-records#customizing-data-before-filling-the-form) and [customizing the saving process](https://filamentphp.com/docs/2.x/admin/resources/editing-records#customizing-data-before-saving).

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

**Reply by kolaente** [reply]
That looks a lot more elegant. Thanks!

**Reply by danharrin** [reply]
It works with other layout components like Grid or Group if you don't want a Fieldset

**Reply by kolaente** [reply]
Yeah I've used Grid, works great
